composer/utils/object_store/gcs_object_store.py

```diff
@@ -115,6 +117,9 @@ def get_object_size(self, object_name: str) -> int:
             FileNotFoundError: If the specified object does not exist in the cloud storage bucket.
             Exception: If an error occurs while trying to retrieve the object's size.
         """
+        if not self.use_gcs_sdk and self.s3_object_store is not None:
```

Comment:
why do you need the s3 none check? shouldn't it always be not none if use sdk is false?

Reply:
(same comment in all places)

Reply:
to please the pyright gods!

Reply:
ah, can you assert instead? otherwise it makes me worried about what happens if s3 object store is none cause it seems like thats possible